### ISSUE TITLE
Broken 3D geometry and other rendering in Asphalt 8: Airborne

### ISSUE THREAD

**Issue by hxdeci** [doitsujin/dxvk#4550]
### Software information

Asphalt 8 Airborne, installed via Heroic Games Launcher
defaulted to max settings and attempting to set lower results in a completely blank screen
performance was, good
results varied between game runs but the geometry artifacts were consistent 

on Windows, half of the 3D rendered screen was stretched

Linux (Debian 12, RX6600)
![image](https://github.com/user-attachments/assets/1a3ce5a6-f82e-4f50-ab00-723cbbd11606)
![image](https://github.com/user-attachments/assets/8f73a2ba-dad4-4533-b315-bbd907180acc)
![Screenshot from 2024-12-24 14-04-35](https://github.com/user-attachments/assets/241b284c-a305-429d-a765-ea982d1a0795)
![Screenshot from 2024-12-24 14-04-47](https://github.com/user-attachments/assets/880d842b-2a8c-48a6-b783-a4bfd0df12f3)
![image](https://github.com/user-attachments/assets/2a682bb6-1249-4a13-9dab-020ed7ccbd00)

Windows (21H2, WX3100):
![image](https://github.com/user-attachments/assets/e839fc7d-f8d1-4acd-87a4-582699d95b4d)

### System information
Linux:
- GPU: Radeon RX 6600
- Driver: AMDGPU, Mesa `24.2.4-1~bpo12+1`, Vulkan 1.3.289
- Wine version: 10.0-rc3 and Wine-GE 8-26 were both tested
- DXVK version: 2.5.2

Windows:
- GPU: Radeon PRO WX 3100
- Driver: Adrenalin 24.9.1
- DXVK Version: 2.5.2

### Apitrace file(s) (LINUX ONLY)
https://drive.google.com/file/d/10IELKDUL8sRjkGIL2cziFSwHKKo5hHoH/view?usp=drive_link
### Log files (Linux only)
[7580986b33344aeb8fd733caa7457a72-lastPlay.log](https://github.com/user-attachments/files/18241636/7580986b33344aeb8fd733caa7457a72-lastPlay.log)

i have **not** tested behavior under WineD3D as i couldnt find a way to get heroic to do so
the above files are from multiple different game sessions that experienced the same behavior, apologies if this inhibits the ability to debug!

**Comment by K0bin** [comment]
It seems to use Wines D3DCompiler so I assume it's a problem with that.

**Comment by Blisto91** [comment]
Is the last screenshot Windows with dxvk? I haven't been able to reproduce that on Windows with Intel or AMD

**Comment by hxdeci** [comment]
> Is the last screenshot Windows with dxvk? I haven't been able to reproduce that on Windows with Intel or AMD
yes. though it doesn't have the same issue as linux in any way... i only opened the game on windows to see if it was the same behavior, _could_ be a driver issue, dont have an nvidia card ready to test

**Comment by doitsujin** [comment]
Are we certain that the Linux issues aren't just missing libraries again (d3dcompiler, d3dx11)? The Windows screenshot looks correct if we ignore the diagonal line, which is an ancient AMD driver bug that never got fixed for old cards.

**Comment by mmbossoni** [comment]
Installed d3dcompiler_47 and _43
Still corrupted

using wined3d fixes the bugs on the menu apparently

**Comment by Blisto91** [comment]
The game caches its shaders so if you've ran it with Wines builtin d3dcompiler_47 first before installing the native version you'll have to wipe `drive_c/users/(username)/AppData/Local/Gameloft/Asphalt8/(random id)/temp/` in your Wine prefix to let the game generate them again.
